lang/typescript/src/utils/address-fields.ts

```diff
@@ -71,3 +71,26 @@ export function splitAddressField(
 
   return parsedAddressObject;
 }
+
+/**
+ * Utility function that attempts to parse an address string based on a regex
+ *
+ * @param fieldDefinition Array of definitions of address sub-fields
+ * @param regexPatterns Regex patterns for parsing sub-fields from an address string
+ * @param address Address string to parse
+ * @returns Partial Address object of fields parsed from string
+ */
+export function regexSplitAddressField(
```

Comment:
Can you also add a test for this function in `address-fields.test.ts`?

Reply:
Done!